### PR TITLE
Fix/read write modes

### PR DESCRIPTION
**Pull request by chrpetre**
this PR implements changes to use numpy for
- read/write binary files (snapshots, svd basis,...)
- read settings file and associated named selection ids
- modify the examples to show more performant code for reading binary files
@EDCarman FYI

### CODE REVIEW

File: examples/evaluate/06-TBROM_input_field.py [hunk 1]
```diff
@@ -129,16 +129,10 @@ def plot_result_comparison(results: pd.DataFrame):
...
-
-    norm = []
-    for i in range(0, int(len(field) / 3)):
-        x = field[i * 3]
-        y = field[i * 3 + 1]
-        z = field[i * 3 + 2]
-        norm.append(math.sqrt(x * x + y * y + z * z))
-    return norm
+    vec = field.reshape((-1, 3))
+    return np.sqrt((vec * vec).sum(axis=1))
```
Comment (lboucin) [inline examples/evaluate/06-TBROM_input_field.py:135]: just FYI, `np.linalg.norm(vec)` does the same.

**Review by lboucin — APPROVED** [review]
GREAT one @chrpetre! Didn't know the `np.fromfile(...)`. Thanks!

File: src/ansys/pytwin/evaluate/tbrom.py [hunk 2]
```diff
@@ -229,15 +215,16 @@ def _read_settings(filepath):
 
         # Create list of name selections indexes
         for name, idsList in namedselection.items():
+            idsListNp = np.array(idsList)
+            ind = np.where(idsListNp == -1)
             finallist = []
```
Comment (EDCarman) [inline src/ansys/pytwin/evaluate/tbrom.py:220]: We can remove finallist since it's no longer used

File: src/ansys/pytwin/evaluate/tbrom.py [hunk 3]
```diff
@@ -178,31 +176,19 @@ def _read_basis(filepath):
...
             nb_val = struct.unpack("Q", f.read(8))[0]
             nb_mc = struct.unpack("Q", f.read(8))[0]
-            basis = []
-            for i in range(nb_mc):
-                vec = []
-                for j in range(nb_val):
-                    vec.append(struct.unpack("d", f.read(8))[0])
-                basis.append(vec)
-        return basis
+        return np.fromfile(filepath, dtype=np.double, offset=32).reshape(-1, nb_val)
```
Comment (EDCarman) [inline src/ansys/pytwin/evaluate/tbrom.py:179]: Not 100% sure how `np.fromfile` works (great find, btw), but it looks like we could also pass is an open file object, so maybe use
```python
with open(filepath, "rb") as f:
    ...
    return np.fromfile(f, dtype=np.double, offset=0).reshape(-1, nb_val)
```
We could then avoid opening the file twice.

File: src/ansys/pytwin/evaluate/tbrom.py [hunk 4]
```diff
@@ -120,10 +118,11 @@ def generate_snapshot(self, on_disk: bool, output_file_path: str, named_selectio
             vec = vec + mc[i] * mnp
         if named_selection is not None:
             pointsids = self.named_selection_indexes(named_selection)
-            listids = []
-            for i in pointsids:
-                for k in range(0, self.field_output_dim):
-                    listids.append(i * self.field_output_dim + k)
+            if self.field_output_dim > 1:
```
Comment (EDCarman) [inline src/ansys/pytwin/evaluate/tbrom.py:121]: I think this might fail for scalar outputs (`dimensionality = 1`), since `listids` is then not defined

**Review by EDCarman — CHANGES_REQUESTED** [review]
Looks good. Just a couple of comments.

File: src/ansys/pytwin/evaluate/tbrom.py (hunk 2 above)
Comment (chrpetre) [reply src/ansys/pytwin/evaluate/tbrom.py:220]: yes indeed, thanks

File: src/ansys/pytwin/evaluate/tbrom.py (hunk 4 above)
Comment (chrpetre) [reply src/ansys/pytwin/evaluate/tbrom.py:121]: good catch Ed, thanks

**Comment by chrpetre** [comment]
thanks for the detailed review @EDCarman, these should be addressed

**Review by EDCarman — APPROVED** [review]

**Pull request merged**